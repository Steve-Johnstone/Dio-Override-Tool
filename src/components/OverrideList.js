import Override from './Override';
import { filterOverrides } from '../helpers/filterOverrides';
import PropTypes from 'prop-types';

const OverrideList = ({
	overrideList,
	searchTerm,
	selectedOverrides,
	handleClick,
}) => {
	return (
		<div className='container mt-5 ml-5'>
			<h2>All Overrides</h2>
			<ul className='override-list' data-testid='override-list'>
				{overrideList
					.filter((override) => filterOverrides(override, searchTerm))
					.map((override) => {
						return (
							<Override
								override={override}
								selectedOverrides={selectedOverrides}
								handleClick={handleClick}
							/>
						);
					})}
			</ul>
		</div>
	);
};

OverrideList.propTypes = {
	overrideList: PropTypes.array,
	searchTerm: PropTypes.string,
	selectedOverrides: PropTypes.array,
	handleClick: PropTypes.func,
};

export default OverrideList;
