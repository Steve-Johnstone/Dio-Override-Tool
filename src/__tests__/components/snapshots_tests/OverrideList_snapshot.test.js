import renderer from 'react-test-renderer';
import OverrideList from '../../../components/UI/OverrideList';

describe('OverrideList component', () => {
	const handleClick = jest.fn();

	it('should display the DOM correctly', () => {
		const tree = renderer
			.create(
				<OverrideList
					overrideList={[
						'Australian Print Receipt',
						'Consolidated Amenities',
						'Event Widget',
						'Insurance Details',
						'Multi Room',
						'No Cancel Button',
						'No Guest Reviews',
						'United Only Points',
					]}
					searchTerm='searching'
					selectedOverrides={[
						'Australian Print Receipt',
						'Consolidated Amenities',
					]}
					handleClick={handleClick}
				/>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
