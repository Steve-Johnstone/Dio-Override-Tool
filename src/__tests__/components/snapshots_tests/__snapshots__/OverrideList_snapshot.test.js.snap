// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverrideList component should display the DOM correctly 1`] = `
<div
  className="container mt-5 ml-5"
>
  <h2>
    All Overrides
  </h2>
  <ul
    className="override-list"
    data-testid="override-list"
  />
</div>
`;
